src/app/components/apparatus-entry/apparatus-entry.component.ts

```diff
@@ -1,26 +1,59 @@
-import { Component, Input } from '@angular/core';
-import { map } from 'rxjs/operators';
-import { ApparatusEntry } from '../../models/evt-models';
+import { ChangeDetectionStrategy, Component, Input, OnInit, Optional, SkipSelf } from '@angular/core';
+import { map, shareReplay } from 'rxjs/operators';
+import { ApparatusEntry, GenericElement } from '../../models/evt-models';
 import { register } from '../../services/component-register.service';
 import { EVTModelService } from '../../services/evt-model.service';
-import { EditionlevelSusceptible } from '../components-mixins';
-
+import { ApparatusEntryDetailComponent } from './apparatus-entry-detail/apparatus-entry-detail.component';
 @Component({
   selector: 'evt-apparatus-entry',
   templateUrl: './apparatus-entry.component.html',
   styleUrls: ['./apparatus-entry.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 @register(ApparatusEntry)
-export class ApparatusEntryComponent extends EditionlevelSusceptible {
+export class ApparatusEntryComponent implements OnInit {
   @Input() data: ApparatusEntry;
 
+  public opened = false;
+  public isInsideAppDetail: boolean;
+  public nestedApps: ApparatusEntry[] = [];
+
   variance$ = this.evtModelService.appVariance$.pipe(
     map((variances) => variances[this.data.id]),
+    shareReplay(1),
   );
 
   constructor(
     private evtModelService: EVTModelService,
+    @Optional() @SkipSelf() private parentDetailComponent?: ApparatusEntryDetailComponent,
   ) {
-    super();
+    this.isInsideAppDetail = this.parentDetailComponent ? true : false;
+  }
+
+  ngOnInit() {
+    if (this.data.hasNestedApp) {
+      this.getNestedApps(this.data);
+    }
+  }
+
+  getNestedApps(app: ApparatusEntry) {
+    const nesApps = app.lemma.content.filter((c: ApparatusEntry | GenericElement) => c.type === ApparatusEntry);
+    nesApps.forEach((nesApp: ApparatusEntry) => {
+      this.nestedApps = this.nestedApps.concat(nesApp);
+      if (nesApp.hasNestedApp) {
+        this.getNestedApps(nesApp);
+      }
+    });
+  }
+
+  toggleAppEntryBox(e: MouseEvent) {
+    e.stopPropagation();
+    this.opened = !this.opened;
+  }
+
+  closeAppEntryBox() {
+    if (this.opened) {
```

Comment:
you can just set it to false since if this.opened is true it will change to false, otherwise it will remain false